### ISSUE TITLE
Infer type from single allOf $ref

### ISSUE THREAD

**Issue by BenjaminPelletier** [Rebilly/ReDoc#915]
Specifying the description for a property with a complex type is [not allowed](https://github.com/OAI/OpenAPI-Specification/issues/556); for example, this is invalid because the description will be wiped out by $ref replacement:
```
FooOperationURL:
    description: Endpoint that provides the ability to perform a foo operation
    type: string
```
```
preview_url:
    description: Endpoint at which to preview data
    $ref: '#/components/schemas/FooOperationURL
```
The suggested (messy & ugly, IMO) workaround for this is to specify the type in an allOf clause:
```
preview_url:
    description: Endpoint at which to preview data
    allOf:
        - $ref: '#/components/schemas/FooOperationURL
```
...but, ReDoc indicates that the type of `preview_url` in the latter case is merely `string` while in the former case it was much more usefully `string (FooOperationURL)`.

I suggest that ReDoc treat a singular allOf $ref as the implied type of the property and therefore print that type hint in the documentation.

**Comment by Tecktron** [comment]
I agree, ReDoc is not handling the `allOf` properly correctly. It should expand every item inside within it including any references. This especially holds true in this case since this is the JSON Schema's suggested way of adding a description to a referenced object and is a very common use case for having reusable definitions.

**Comment by RomanHotsiy** [comment]
Until this is fixed you can try using the following workaround:

```yaml
preview_url:
  allOf:
    - $ref: '#/components/schemas/FooOperationURL
    - description: Endpoint at which to preview data
```

**Comment by mattiasrunge** [comment]
For me it seems the workaround with allOf does not work correctly without adding type and then not for array types.
Since we see this as a limitation of the specification, could it not be possible to support this case with an extension.

```yml
preview_url:
    x-description: Endpoint at which to preview data
    $ref: #/components/schemas/FooOperationURL
```
